lib/node/request.js

```diff
@@ -32,22 +33,36 @@ class Request {
     this._url = url;
   }
 
+  setAgent(agent) {
+    this._agent = agent;
+  }
+
   setRequestHeader(key, value) {
     this._headers[key] = value;
   }
 
   send(body) {
     let options = parse(this._url);
+    options.agent = this._agent;
     options.method = this._method;
     options.headers = this._headers;
     if (body && body.size) options.headers["Content-Length"] = body.size;
 
+    // sanity check...
+    if (this._agent instanceof Object && this._agent.protocol != options.protocol) {
+      const err = `Cannot use ${this._agent.protocol} Agent for ${options.protocol} request`;
+      this.onerror(new Error(err));
+      return;
+    }
+
     let req = this._request = (options.protocol !== "https:")? http.request(options) : https.request(options);
     req.on("response", (res) => {
       this.status = res.statusCode;
       this._resHeaders = res.headers;
 
       this.onload();
+
+      res.resume();   // this allows Agent to reuse same connection for next HTTP request
```

Comment:
Please link to https://github.com/nodejs/node/issues/9595 in order to explain this statement.

Reply:
Adding link in code comment - not sure this is what you meant.